packages/hub/card-services.ts

```diff
@@ -0,0 +1,233 @@
+import Session from '@cardstack/plugin-utils/session';
+import { declareInjections } from '@cardstack/di';
+import { todo } from '@cardstack/plugin-utils/todo-any';
+import { SingleResourceDoc, ResourceObject, ResourceIdentifierObject, CollectionResourceDoc } from 'jsonapi-typescript';
+import { get, uniqBy } from 'lodash';
+
+const cardIdDelim = '::';
+const cardBrowserAssetFields = [
+  'isolated-template',
+  'isolated-js',
+  'isolated-css',
+  'embedded-template',
+  'embedded-js',
+  'embedded-css',
+  'edit-template',
+  'edit-js',
+  'edit-css',
+];
+interface CardContext {
+  repository: string;
+  packageName: string;
+  cardId?: string;
+  modelId?: string;
+}
+
+export = declareInjections({
+  writers: 'hub:writers',
+  indexers: 'hub:indexers',
+  searchers: 'hub:searchers',
+  currentSchema: 'hub:current-schema',
+},
+
+class CardServices {
+  writers: todo;
+  indexers: todo;
+  searchers: todo;
+  currentSchema: todo;
+
+  cardContextFromId(id: string) {
```

Comment:
Can this and `cardContextToId` be private?

Reply:
actually i pulled it put into plugin-helpers a subsequent PR as I needed it in a few other places (specifically around field ID `::` handling)

Reply:
Yeah, I see that, but that is just another reason I don't think we should be doing that field id handling where we're doing it.